Tests/Unit/NetworkServer/ModuleConnectionHostTest.cs

```diff
@@ -273,10 +274,79 @@ public async Task OnDirectMethodCall_ClearCache_When_Correct_Should_Work()
             networkServerConfiguration.IoTEdgeTimeout = 5;
 
             await using var moduleClient = new ModuleConnectionHost(networkServerConfiguration, classCMessageSender.Object, this.loRaModuleClientFactory.Object, loRaDeviceRegistry.Object, loRaDeviceApiServiceBase, NullLogger<ModuleConnectionHost>.Instance, TestMeter.Instance);
-            await moduleClient.OnDirectMethodCalled(new Microsoft.Azure.Devices.Client.MethodRequest(Constants.CloudToDeviceClearCache), null);
+            await moduleClient.OnDirectMethodCalled(new MethodRequest(Constants.CloudToDeviceClearCache), null);
             loRaDeviceRegistry.VerifyAll();
         }
 
+        [Fact]
+        public async Task OnDirectMethodCall_DropConnection_Should_Work_As_Expected()
+        {
+            // arrange
+            var networkServerConfiguration = new NetworkServerConfiguration();
+            var classCMessageSender = new Mock<IClassCDeviceMessageSender>(MockBehavior.Strict);
+            var loRaDeviceRegistry = new Mock<ILoRaDeviceRegistry>(MockBehavior.Strict);
+            var devEui = new DevEui(0);
+            var mockedDevice = new Mock<LoRaDevice>(null, devEui, null);
+            _ = loRaDeviceRegistry.Setup(x => x.GetDeviceByDevEUIAsync(devEui)).ReturnsAsync(mockedDevice.Object);
+            var c2d = $"{{\"DevEui\": \"{ devEui }\", \"Fport\": 1, \"MessageId\": \"{ Guid.NewGuid() }\"}}";
+
+            // act
+            await using var moduleClient = new ModuleConnectionHost(networkServerConfiguration, classCMessageSender.Object, this.loRaModuleClientFactory.Object, loRaDeviceRegistry.Object, loRaDeviceApiServiceBase, NullLogger<ModuleConnectionHost>.Instance, TestMeter.Instance);
+            await moduleClient.OnDirectMethodCalled(new MethodRequest(Constants.CloudToDeviceCloseConnection, Encoding.UTF8.GetBytes(c2d)), null);
+
+            // assert
+            loRaDeviceRegistry.VerifyAll();
+            mockedDevice.Verify(x => x.CloseConnectionAsync(It.IsAny<CancellationToken>(), true), Times.Once);
+        }
+
+        public static TheoryData<byte[]> DropConnectionInvalidMessages =>
+            new TheoryData<byte[]>
+                {
+                    null,
+                    Array.Empty<byte>(),
+                    Encoding.UTF8.GetBytes("{{\"DevEui\": null, \"Fport\": 1}}"), // invalid DevEui
+                    Encoding.UTF8.GetBytes("{{\"DevEui\": \"{ new DevEui(0) }\", \"Fport\": 1, \"MessageId\": 123}}") // invalid message id
```

Comment:
Did you mean to have an interpolated string here?

```suggestion
                Encoding.UTF8.GetBytes($"{{\"DevEui\": \"{ new DevEui(0) }\", \"Fport\": 1, \"MessageId\": 123}}")); // invalid message id
```

What concerns me further is that the tests are passing nonetheless even though the theory data is completely invalid JSON in the first place. Unfortunately, the reliance on just the status code being `HttpStatusCode.BadRequest` means that once cannot know if the individual cases of invalid `DevEui` or `MessageId` are really being exercised. I would recommend to make the tests break before making them go green and relying on another strategy (either returning something in the response of sniffing the log).

I also think the theory data should be simple JSON strings and the conversion to bytes be done within the test method. The problem with bytes is that it's not easy to discern in the text UI what's being tested:

```
OnDirectMethodCall_DropConnection_Should_Return_Bad_Request_When_Invalid_Message(c2d: [123, 123, 34, 68, 101, ...])
OnDirectMethodCall_DropConnection_Should_Return_Bad_Request_When_Invalid_Message(c2d: [123, 123, 34, 68, 101, ...])
OnDirectMethodCall_DropConnection_Should_Return_Bad_Request_When_Invalid_Message(c2d: [])
OnDirectMethodCall_DropConnection_Should_Return_Bad_Request_When_Invalid_Message(c2d: null)
```

Reply:
Good catch, thanks! Fixed with 0be1f5b3cda6ab3abbfb0bb1b65006befb7139a5